src/Sylius/Bundle/CoreBundle/Validator/Constraints/UniqueReviewerEmailValidator.php

```diff
@@ -56,8 +56,12 @@ public function validate($value, Constraint $constraint): void
         }
     }
 
-    private function getAuthenticatedUserEmail(TokenInterface $token): ?string
+    private function getAuthenticatedUserEmail(?TokenInterface $token): ?string
     {
+        if (null === $token) {
+            return null;
+        }
```

Comment:
IMO It could be moved after the line `$token = $this->tokenStorage->getToken();` and if token is `null` we can instantly return nothing in the `validate` method.

Reply:
Alternatively, move `$token = $this->tokenStorage->getToken();` inside `getAuthenticatedUserEmail` itself since the token is not used for anything else.
Either way, I see a bunch of problems with this validator, refactoring it would be a good idea, but that's outside the scope of this PR.